File: Backend_notes/src/models/Note.js
const mongoose = require('mongoose');

const noteSchema = mongoose.Schema({
    id: {
        type: String,
        unique: true,
        required: true
    },
    userid: {
        type: String,
        required: true,
    },
    title: {
        type: String,
        required: true,
    },
    content: {
        type: String,
    },
    dateadded: {
        type: Date,
        default: Date.now,
    },
    type: {
        type: String,
    }
});

module.exports = mongoose.model("Note", noteSchema);

File: lib/models/note.dart
class Note {
  String? id;
  String? userid;
  String? title;
  String? content;
  DateTime? dateadded;
  String? type;

  Note({
    this.id,
    this.userid,
    this.title,
    this.content,
    this.dateadded,
    this.type,
  });

  factory Note.fromMap(Map<String, dynamic> map) {
    return Note(
      id: map["id"],
      userid: map["userid"],
      title: map["title"],
      content: map["content"],
      dateadded: DateTime.tryParse(map["dateadded"]),
      type: map["type"],
    );
  }

  Map<String, dynamic> toMap() {
    return {
      "id": id,
      "userid": userid,
      "title": title,
      "content": content,
      "dateadded": dateadded!.toIso8601String(),
      "type": type,
    };
  }
}
